### PR TITLE
ENH Improve component focusing

### PR DESCRIPTION
**Pull request by emteknetnz**
Issue https://github.com/silverstripe/silverstripe-linkfield/issues/245

This PR:
- Makes focus when using mouse and keyboard consistent. Previously only using the keyboard has special logic for setting the focus state
- Fixes a timing issue with focus when using a keyboard
- Swaps around the tab order of the drag handle so that it is last rather than first

### CODE REVIEW

File: client/src/components/LinkModal/LinkModal.js [hunk 1]
```diff
@@ -86,7 +86,6 @@ LinkModal.propTypes = {
   isOpen: PropTypes.bool.isRequired,
   onSuccess: PropTypes.func.isRequired,
   onClosed: PropTypes.func.isRequired,
-  autoFocus: PropTypes.bool,
```
Comment (ChloeSartorelli) [inline client/src/components/LinkModal/LinkModal.js:89]: This prop is still being deconstructed from props in line 32

**Review by GuySartorelli — CHANGES_REQUESTED** [review]
When I archive a link and then reorder links via the keyboard I get this error in the console:
> Assertion failed: Input argument is not an HTMLInputElement

I don't get that if I reorder with the mouse.

**Comment by GuySartorelli** [comment]
Other than the specific problems above, it works great locally.

File: client/src/components/LinkModal/LinkModal.js (hunk 1 above)
Comment (emteknetnz) [reply client/src/components/LinkModal/LinkModal.js:89]: I've removed it from line 32, it was unused

**Comment by emteknetnz** [comment]
> When I archive a link and then reorder links via the keyboard I get this error in the console:

I'm unable to replicate this. Could you please double check this is still happening with the latest version of this PR and if so then demo it to me.

**Comment by GuySartorelli** [comment]
Can't replicate that problem anymore, no idea what was causing it.
LGTM but has merge conflicts.

**Comment by emteknetnz** [comment]
Done

**Review by GuySartorelli — APPROVED** [review]
LGTM

**Pull request merged**